### ISSUE TITLE
Showing negative cost if I restarted app which were running from last one month aprox.

### ISSUE THREAD

**Issue by cfmagnum** [cloudfoundry-incubator/cf-abacus#709]
I tested one scenario:

I have an org in which 3 apps are running from last one month. Now I deployed abacus on CF. 

After some time I restarted all three apps and checked the cost of that org. **It's showing negative cost.**

If this is the case, how can I start to use Abacus on production where many apps are already running. And will be restarted anytime later but don't when.

**Comment by hsiliev** [comment]
The scenario you describe results in 1 stop & 1 start event. Bridge & `linux-container` plan require start event as the first event for an app, otherwise we end up with negative cost exactly as you noticed.

To start metering you need to start a new billing epoch by purging the events and repopulating the CC DB with this call: http://apidocs.cloudfoundry.org/latest-release/app_usage_events/purge_and_reseed_app_usage_events.html

This will clear the CC DB from all old events and repopulate it with start events for currently running apps.

**Comment by cfmagnum** [comment]
@hsiliev Thanks. I tried the approach you suggested above. It's worked fine. And started the billing from when I deployed abacus bridge.

But after 2 days I redeployed all abacus application with updated code. At that time as well I called the  
/app_usage_events/purge_and_reseed_app_usage_events api before starting the cf-abacus-bridge.
After successful deployment. I checked the billing report. It was showing negative charge.

My expectation is, it should continue the calculations ahead from the cost before redeploy. But it's showing negative value after redeploy. 

Can you please help me to resolve this.

**Comment by hsiliev** [comment]
@cfmagnum purge should be called only once in the lifetime of CF deployment - shortly before Abacus bridge & renewer are deployed. Calling it more than once is not supported since the purge call rewrites the events DB.

**Comment by cfmagnum** [comment]
@hsiliev Thanks. I have done the same and now that issue has been resolved. It's showing the positive cost.

But found another issue.

Scenario: 
    - I pushed two apps and then deployed abacus along with renewer
    - A positive cost was reflected.
    - Then I stopped an application.
    - And the cost value in the minute window for that particular application appeared negative.
    - Some time I have noticed the -ve cost in hrs window as well.

Can you please let me know what's wrong.

**Comment by hsiliev** [comment]
I *think* that the negative cost is due to the fact that in this time-window we have only the stop event. 
@KRuelY can this be the case?

**Comment by hsiliev** [comment]
Negative value can appear in windows if:
* start was in other cell (for minutes, hours). Start is in previous minute and stop is in this minute
* got stop event, before start. Can happen if we have lots of events flowing and some was retried

We should not have charge and summary produce negative results once all events are processed. Moving from windows to days & month slots can help here and we are considering to do so for v2
